model-api/src/commonMain/kotlin/org/modelix/model/api/INode.kt

```diff
@@ -47,6 +47,15 @@ interface INode {
 
     fun tryGetConcept(): IConcept? = getConceptReference()?.tryResolve()
 
+    /**
+     * Replaces this node with a new node that uses the same id as this node.
+     * Properties, references and children will be the same.
+     *
+     * @param concept the concept of the new node
+     * @return replacement for this node
+     */
+    fun replaceNode(concept: ConceptReference): INode
```

Comment:
The name and the documentation somehow lack the important aspect that this is intended to be used for changing concepts. Maybe something like these names could work better:

* changeConcept
* replaceWithNewConcept

Reply:
I updated the documentation. The name is the one we agreed on during refinement.